src/client.rs

```diff
@@ -158,13 +238,7 @@ impl Client {
 
     /// Create a corresponding object of an [Index] without any check or doing an HTTP call.
     pub fn index(&self, uid: impl Into<String>) -> Index {
-        Index {
-            uid: Arc::new(uid.into()),
-            client: self.clone(),
-            primary_key: None,
-            created_at: None,
-            updated_at: None,
-        }
+        Index::new(uid, self.clone())
```

Comment:
It is the `Clone` I ask for the Key handler?

Reply:
@irevoire I'm not sure I remember exactly why we had to do this

Reply:
> It is the Clone I ask for the Key handler?

No, it's the `Client` that's being cloned not the key handler.

We had to do this because our methods take `&self: &Client` as a parameter, and `Index::new` takes a `Client` directly, thus we clone the reference and get a real `Client` back.